tests/trace_server/test_calls_query_builder.py

```diff
@@ -1012,8 +1005,9 @@ def test_calls_query_with_or_between_start_and_end_fields() -> None:
             any(calls_merged.inputs_dump) AS inputs_dump,
             any(calls_merged.output_dump) AS output_dump
         FROM calls_merged
-        WHERE
-            calls_merged.project_id = {pb_4:String}
+        WHERE calls_merged.project_id = {pb_6:String}
+            AND (((calls_merged.inputs_dump LIKE {pb_4:String} OR calls_merged.inputs_dump IS NULL)
```

Comment:
so this was completely incorrect before?

Reply:
We basically flattened nested conditions, no we retain the nested structure, which requires checking null at every reference @tssweeney

Reply:
oh, sorry, in this case we just ejected when we saw an OR operation, so we _dont_ do the optimization currently. Now we can do the optimization!